x/poolmanager/router_test.go

```diff
@@ -1990,3 +1990,79 @@ func (suite *KeeperTestSuite) TestSplitRouteExactAmountOut() {
 		})
 	}
 }
+
+func (s *KeeperTestSuite) TestGetTotalPoolLiquidity() {
+	var (
+		defaultPoolCoinOne = sdk.NewCoin("usdc", sdk.OneInt())
+		defaultPoolCoinTwo = sdk.NewCoin("eth", sdk.NewInt(2))
+		nonPoolCool        = sdk.NewCoin("uosmo", sdk.NewInt(3))
+
+		defaultCoins = sdk.NewCoins(defaultPoolCoinOne, defaultPoolCoinTwo)
+	)
+
+	tests := []struct {
+		name           string
+		poolId         uint64
+		poolLiquidity  sdk.Coins
+		expectedResult sdk.Coins
+		expectedErr    error
+	}{
+		{
+			name:           "valid with 2 coins",
+			poolId:         1,
+			poolLiquidity:  defaultCoins,
+			expectedResult: defaultCoins,
+		},
+		{
+			name:           "valid with 1 coin",
+			poolId:         1,
+			poolLiquidity:  sdk.NewCoins(defaultPoolCoinTwo),
+			expectedResult: sdk.NewCoins(defaultPoolCoinTwo),
+		},
+		{
+			// can only happen if someone sends extra tokens to pool
+			// address. Should not occur in practice.
+			name:           "valid with 3 coins",
+			poolId:         1,
+			poolLiquidity:  sdk.NewCoins(defaultPoolCoinTwo, defaultPoolCoinOne, nonPoolCool),
+			expectedResult: defaultCoins,
+		},
+		{
+			// this can happen if someone sends random dust to pool address.
+			name:           "only non-pool coin - does not show up in result",
+			poolId:         1,
+			poolLiquidity:  sdk.NewCoins(nonPoolCool),
+			expectedResult: sdk.Coins(nil),
+		},
```

Comment:
Some of these tests are only relevant in the context of concentrated liquidity (looks like this is where they were copied from).

What makes sense to be tested here:
- can route to CL module and get liquidity
- can route to gamm module and get liquidity
- some error cases specific to poolmanager

Please refactor the tests to be relevant for this level of abstraction

Reply:
added